libraries/WiFi/src/WiFiClient.cpp

```diff
@@ -210,22 +215,34 @@ int WiFiClient::connect(IPAddress ip, uint16_t port)
 {
     return connect(ip,port,_timeout);
 }
+
 int WiFiClient::connect(IPAddress ip, uint16_t port, int32_t timeout)
 {
+    struct sockaddr_storage serveraddr;
     _timeout = timeout;
-    int sockfd = socket(AF_INET, SOCK_STREAM, 0);
+    int sockfd = -1;
+
+    if (ip.type() == IPv6) {
+        struct sockaddr_in6 *tmpaddr = (struct sockaddr_in6 *)&serveraddr;
+        memset((char *) tmpaddr, 0, sizeof(struct sockaddr_in));
```

Comment:
Shouldn't the size be `sizeof(sockaddr_in6)`?

Reply:
Actually you can remove both `memset` by simply declaring

`struct sockaddr_storage serveraddr = {};`

which forces the compiler to initialize the structure with zeros